include/fmt/format.h

```diff
@@ -2137,8 +2274,12 @@ FMT_CONSTEXPR void parse_format_string(
       if (begin == end) return;
       for (;;) {
         const Char *p = FMT_NULL;
-        if (!find<IS_CONSTEXPR>(begin, end, '}', p))
-          return handler_.on_text(begin, end);
+        if (!find<IS_CONSTEXPR>(begin, end, '}', p)) {
+          if (begin != end) {
+            handler_.on_text(begin, end);
+          }
+          return;
+        }
```

Comment:
This looks unrelated.

Reply:
Parts preparation uses the `parse_format_string()` function to count how many parts the format has. Without the `if (begin != end)`, the `handler_.on_text()` is called, even if `begin == end` and that causes creation of an additional, empty format part. With the change, prepared format has one part less.